libnetwork/netutils/utils_linux.go

```diff
@@ -24,6 +24,7 @@ var (
 
 // CheckRouteOverlaps checks whether the passed network overlaps with any existing routes
 func CheckRouteOverlaps(toCheck *net.IPNet) error {
+	networkGetRoutesFct := networkGetRoutesFct
```

Comment:
This only affects our unit-tests, or could this be something that's somehow hit in actual (non-test) situations?

Reply:
`ns.NlHandle()` will always return the same value until the next time `libnetwork/ns.Init()` is called. Outside of tests `ns.Init()` is only called through a `sync.Once` so this only affects unit tests.